Assets/Scripts/Item.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "New Item", menuName = "Inventory/Item")]
+public class Item : ScriptableObject
+{
+    [SerializeField]
+    new private string name = "New Item";
+    [SerializeField]
+    private Sprite icon = null;
```

Comment:
I guess those fields should be public. Otherwise the inventory can't access the correct sprite and name.
FYI: Default value "null" is always the case, so no need to define it with it. If we decide to always define fields, then it's fine, but those values will always be overwritten by the values from the inspector which can be a bit confusing.

Reply:
You're right. Or Getter/Setter for each field. Which one do you prefer?

Reply:
Probably a getter only with a property:
`public Sprite Icon => icon;`
Also it's nicer to give the file the specific item name and don't use an extra one. Easier to find later on

Reply:
> Also it's nicer to give the file the specific item name and don't use an extra one. Easier to find later on

The idea was that this item script is for general purpose.
Each specific item (such as wood) is stored under the Items folder. Is that what you meant?

Reply:
I meant that each ScriptableObject should be named after the item itself. So getting the name of the ScriptableObject itself will give you the item name. So there is no need for a field.

Reply:
The existing field `name`, which you are overriding, contains the name of the actual file which represents the instance of the scriptable object 👍

Reply:
Ah, got it, yeah that's probably better :)